tests/transports/test_all_plugins.py

```diff
@@ -33,16 +30,22 @@
 # TODO : silly cases of copy/put/get from self to self
 
 
+@pytest.fixture(scope='function')
+def remote_tmp_path(tmp_path_factory):
+    """Mock the remote tmp path using tmp_path_factory to create folder start with 'remote'"""
+    return tmp_path_factory.mktemp('remote')
+
+
 @pytest.fixture(scope='function', params=entry_point.get_entry_point_names('aiida.transports'))
-def custom_transport(request, tmp_path, monkeypatch) -> Transport:
+def custom_transport(request, tmp_path_factory, monkeypatch) -> Transport:
     """Fixture that parametrizes over all the registered implementations of the ``CommonRelaxWorkChain``."""
     plugin = TransportFactory(request.param)
 
     if request.param == 'core.ssh':
         kwargs = {'machine': 'localhost', 'timeout': 30, 'load_system_host_keys': True, 'key_policy': 'AutoAddPolicy'}
     elif request.param == 'core.ssh_auto':
         kwargs = {'machine': 'localhost'}
-        filepath_config = tmp_path / 'config'
+        filepath_config = tmp_path_factory.mktemp('transport') / 'config'
```

Comment:
Maybe add a comment for one line, why you did it this way..
probably because `listdir` would also list `'config'` for `core.auto_ssh`